File: back_end/src/main/java/com/example/entity/request/Care.java
package com.example.entity.request;
import com.example.entity.response.CustomerResp;
import com.example.entity.response.DeptResp;
import com.example.entity.response.EmpResp;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

/**
 * @author: liuleimin
 * @date: 2020/10/10
 * @description:
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Care {
    private Integer careId;
    private String careZt;//关怀主题
    private String carelxrcontacts;//联系人
    private Date careData;
    private String careexecutor;//执行人
    private String carenr;//关怀内容
    private String carecustomerfk;
    private Integer cusId;
    private Integer empId;
    private String startDate;
    private String endDate;
    private Integer pageNum;
    private Integer pageSize;
    private CustomerResp customerResp;
    private EmpResp empResp;
    private DeptResp deptResp;
}

File: src/network/system/care.js
import {postJson,get} from "../request";

export const careHttp = {
  add:function (care) {
    let params = {
      url:'/care/add',
      data:{
        'careZt':care.careZt,
        'carelxrcontacts':care.carelxrcontacts,
        'careData':care.careData,
        'careexecutor':care.careexecutor,
        'carenr':care.carenr,
        'carecustomerfk':care.carecustomerfk,
        'cusId':care.cusId,
        'empId':care.empId
      }
    }
    return postJson(params)
  },
  get:function (careId) {
    let params = {
      url:'/care',
      data:{
        'careId':careId
      }
    }
    return get(params)
  },
  update:function (care) {
    let params = {
      url:'/care/update',
      data:{
        'careId':care.careId,
        'careZt':care.careZt,
        'carelxrcontacts':care.carelxrcontacts,
        'careData':care.careData,
        'careexecutor':care.careexecutor,
        'carenr':care.carenr,
        'carecustomerfk':care.carecustomerfk,
        'cusId':care.cusId,
        'empId':care.empId
      }
    }
    return postJson(params)
  },
  del:function (careId) {
    let params = {
      url:'/care/del',
      data:{
        'careId':careId
      }
    }
    return postJson(params)
  },
  list:function (care) {
    let params = {
      url:'/care/list',
      data:{
        'careZt':care.careZt,
        'carelxrcontacts':care.carelxrcontacts,
        'careData':care.careData,
        'careexecutor':care.careexecutor,
        'carenr':care.carenr,
        'carecustomerfk':care.carecustomerfk,
        'cusId':care.cusId,
        'empId':care.empId

      }
    }
    return get(params)
  },
  queryEmp:function (care) {
    let params = {
      url:'/care/list',
      data:{
        'careZt':care.careZt,
        'carelxrcontacts':care.carelxrcontacts,
        'cusId':care.cusId,
        'empId':care.empId,
        'empStatus':care.empStatus,
        'startDate':care.startDate,
        'endDate':care.endDate,
        'careexecutor':care.careexecutor,
        'carenr':care.carenr,
        'carecustomerfk':care.carecustomerfk,
        'pageNum':care.pageNum,
        'pageSize':care.pageSize
      }
    }
    return get(params)
  },
  listPage:function (pageNum,pageSize) {
    let params = {
      url:'/care/list',
      data:{
        'pageNum':pageNum,
        'pageSize':pageSize
      }
    }
    return get(params)
  }
}
